scripts/publish-npm.js

```diff
@@ -126,12 +126,38 @@ if (nightlyBuild) {
   releaseVersion = tagsWithVersion[tagsWithVersion.length - 1].slice(1);
 }
 
-// -------- Generating Android Artifacts with JavaDoc
-if (exec('./gradlew :ReactAndroid:installArchives').code) {
-  echo('Could not generate artifacts');
-  exit(1);
+const buildAndroid = (rebuildOnError) => {
+  // -------- Generating Android Artifacts with JavaDoc
+  if (exec('./gradlew :ReactAndroid:installArchives').code) {
+    echo('Could not generate artifacts.');
+    exit(1);
+  }
+
+  const aarLocation = `android/com/facebook/react/react-native/${releaseVersion}/react-native-${releaseVersion}.aar`;
+  const tmpLocation = `/tmp/react-native-${releaseVersion}`;
+
+  // -------- Remove unzipped Android artifact (if any) 
+  exec(`rm -rf ${tmpLocation}`);
+
+  // -------- Check if Android artifact contains all the files
+  const {stdout: aarContent} = exec(`unzip ${aarLocation} -d ${tmpLocation} | grep libfbjni.so`);
+  if ((aarContent.match(/libfbjni.so/g) || []).length === 4) {
+    echo(`Artifacts validated, continuing.`);
+    return;
+  }
+
+  // -------- Temporary fix for broken Android artifact is to build with Gradle once again
+  if (rebuildOnError) {
+    echo('Artifacts are corrupted. Rebuilding with Gradle once again to fix it.');
```

Comment:
I’d personally reserve ‘corruption’ for cases where the files are present but do not match expected contents. In this case the files are simply missing, so perhaps we can just say, which might be more helpful anyways to readers of this error in the future? E.g.

```suggestion
    echo(`${aarLocation} is missing contents. Rebuilding with Gradle again in an effort to try to fix it.`);
```